### ISSUE TITLE
Consistent JSON Post Results

### ISSUE THREAD

**Issue by stonetip** [WP-API/WP-API#2292]
When retrieving a single post by ID, a JSON object is returned. When retrieving a single post by a slug, a single post as a search result or when retrieving multiple posts a JSON array is returned. An array would be preferred in all cases. As it stands, we now have to pollute what was supposed to be a simple HTTP Request helper class with a check and have it return a specific model rather than a more generic JSON response.

Again, desired behavior (at least when posts exist) is to return an array regardless of the number of posts. I realize this will break implementations that expect a retrieval by ID to be an object. However, in the long run it would provide a unified and consistent response.

**Comment by danielbachhuber** [comment]
You're welcome to change this behavior for your own application of the API. However, we won't be changing the format of an individual resource from an object to an array in the main project.

**Comment by stonetip** [comment]
That doesn't address the inconsistency of retrieving a single post using a slug instead of an ID. How is that also not "an individual resource"? Why is it also different when a single result is returned from a filtered request?

Sure, I can change my code (and already have) but am trying to make you aware that this policy requires workarounds and additional testing. As stated, I fully appreciate the inconvenience changing this might cause. However, in the long term, having a consistent, logical interface is going to make adoption more likely...for any project and for all future users of the WP REST API.

**Comment by danielbachhuber** [comment]
> That doesn't address the inconsistency of retrieving a single post using a slug instead of an ID. How is that also not "an individual resource"?

When you use `GET wp/v2/posts slug==foo` to _filter_ a collection of posts to a single post, you're fundamentally filtering a collection, not accessing an individual resource.

We've discussed the format of our responses for many, many hours, and considered many different types of implementations. We're reasonably happy with the implementation we've settled upon.
